avida-ts/packages/avida-tx-utils/src/upload_and_inst.ts

```diff
@@ -0,0 +1,92 @@
+import { MsgStoreCode, MsgInstantiateContract } from "cosmes/client";
+import {
+  CosmosBaseV1beta1Coin as Coin,
+  CosmosTxV1beta1Fee as Fee,
+  CosmosBaseAbciV1beta1GasInfo as GasInfo,
+} from "cosmes/protobufs";
+
+import { getWallet } from "./wallet";
+import fs from "fs";
+
+import { type UnsignedTx } from "cosmes/wallet";
+
+export async function deploy(
+  chainConfigPath: string,
+  contractPath: string,
+  deployerMnemonic: string,
+  contractInstMsg: { [k: string]: unknown },
+  initFund: Coin[],
+  contractLabel: string,
+  memo: string | undefined = undefined,
+): Promise<string> {
+  console.info(
+    "Deploying contract at: ",
+    contractPath,
+    "\n to chain specified at:",
+    chainConfigPath,
+  );
+
+  const deployer = getWallet(chainConfigPath, deployerMnemonic);
+  console.info("Deployer Addr:", deployer.address); // prints the bech32 address
+
+  // Store wasm MsgStoreCode
+  const wasm: Buffer = fs.readFileSync(contractPath);
+  const wasmByteCode = new Uint8Array(wasm.buffer);
+  console.debug("length of wasmByteCode:", wasmByteCode.length);
+
+  const storeMsg = new MsgStoreCode({
+    wasmByteCode,
+    sender: deployer.address,
+  });
+
+  const storeTx: UnsignedTx = {
+    msgs: [storeMsg],
+    memo,
+  };
+
+  let fee = await deployer.estimateFee(storeTx);
```

Comment:
Are there any possible errors here could happen? Internet issues or others.
I think it worth at least to use `try-catch` construction.

Reply:
fixed in https://github.com/nymlab/AVIDA/pull/6/commits/7ce3c1886296d88d19edcfd650036b81c8dcdf75